src/mpi/romio/adio/ad_gpfs/ad_gpfs_aggrs.c

```diff
@@ -641,8 +645,18 @@ void ADIOI_GPFS_Calc_others_req(ADIO_File fd, int count_my_req_procs,
     int i;
     ADIOI_Access *others_req;
 
+#if MPI_VERSION >= 1234 || (MPI_VERSION == 1234 && MPI_SUBVERSION > 1234)
+#define HAS_ALLTOALLV_C 1
+#else
+#define HAS_ALLTOALLV_C 0
+#endif
     /* Parameters for MPI_Alltoallv */
-    int *scounts, *sdispls, *rcounts, *rdispls;
+    MPI_Aint *sdispls, *rdispls;
+#if HAS_ALLTOALLV_C
+    MPI_Count *scounts, *rcounts;
+#else
+    int *scounts, *rcounts;
+#endif
```

Comment:
Just move the declarations to where it used (Malloc).

Reply:
Good idea, I moved it